src/main/java/fi/aalto/cs/apluscourses/ui/courseproject/CourseSelectionView.java

```diff
@@ -0,0 +1,117 @@
+package fi.aalto.cs.apluscourses.ui.courseproject;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.ui.ColoredListCellRenderer;
+import com.intellij.ui.SimpleTextAttributes;
+import com.intellij.ui.components.JBList;
+import fi.aalto.cs.apluscourses.presentation.CourseItemViewModel;
+import fi.aalto.cs.apluscourses.presentation.CourseSelectionViewModel;
+import fi.aalto.cs.apluscourses.ui.base.OurDialogWrapper;
+import fi.aalto.cs.apluscourses.ui.base.TextField;
+import fi.aalto.cs.apluscourses.utils.PluginResourceBundle;
+import icons.PluginIcons;
+import java.awt.Font;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.ListSelectionModel;
+import javax.swing.SwingConstants;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class CourseSelectionView extends OurDialogWrapper {
+  private CourseSelectionViewModel viewModel;
+
+  private JPanel basePanel;
+  private JLabel urlFieldLabel;
+  private JLabel courseListLabel;
+  private JList<CourseItemViewModel> courseList;
+  private TextField urlField;
+
+  private static final SimpleTextAttributes NAME_TEXT_ATTRIBUTES
+      = new SimpleTextAttributes(SimpleTextAttributes.STYLE_BOLD, null);
+  private static final SimpleTextAttributes SEMESTER_TEXT_ATTRIBUTES
+      = new SimpleTextAttributes(SimpleTextAttributes.STYLE_ITALIC, null);
+
+  /**
+   * Construct a course selection dialog.
+   */
+  public CourseSelectionView(@NotNull Project project,
+                             @NotNull CourseSelectionViewModel viewModel) {
+    super(project);
+    this.viewModel = viewModel;
+    setTitle(PluginResourceBundle.getText("ui.courseProject.courseSelection.title"));
+
+    courseList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+    courseList.addListSelectionListener(listSelectionEvent -> {
+      CourseItemViewModel selected = courseList.getSelectedValue();
+      if (selected != null) {
+        viewModel.selectedCourseUrl.set(selected.getUrl());
+      }
+    });
```

Comment:
This logic could perhaps be moved to view model with `selectedValueBindable` and `selectedCourseProperty`.

Reply:
Agreed, and added a note about this, but merging for now, as we need to release soon